File: Hello_World.py
import time
import resource # used for obtaining memory usage


# Start timing
start_time = time.time()
start_cpu_time = time.process_time()

# Print Hello, World!
print("Hello, World!")

# End timing and calculate times
end_cpu_time = time.process_time()
end_time = time.time()

# Convert times to milliseconds
cpu_time = (end_cpu_time - start_cpu_time) * 1000
exec_time = (end_time - start_time) * 1000

# Print metrics
print(f"CPU Time: {cpu_time:.2f} ms")
print(f"Execution Time: {exec_time:.2f} ms")

# Get peak memory usage. Comment out this block to measure times without the added overhead.
peak_memory = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
print(f"Peak Memory Usage: {peak_memory} KB")

File: Hello_World.cpp
#include <iostream>
#include <chrono>
#include <ctime>
#include <sys/resource.h> // used for obtaining memory usage

/*
A simple C++ program that prints "Hello, World!" and measures execution time.
*/
int main()
{
    // Start timing
    auto start = std::chrono::high_resolution_clock::now();
    std::clock_t c_start = std::clock();

    // Print Hello, World!
    std::cout << "Hello, World!" << std::endl;

    // End timing and calculate execution time
    std::clock_t c_end = std::clock();
    auto end = std::chrono::high_resolution_clock::now();

    // calculate CPU and Execution time in milliseconds
    // CLOCKS_PER_SEC is the number of clock ticks per second
    double cpu_time = 1000.0 * (c_end - c_start) / CLOCKS_PER_SEC;
    std::chrono::duration<double, std::milli> exec_time = end - start;
    std::cout << "Execution time: " << exec_time.count() << " ms" << std::endl;
    std::cout << "CPU time: " << cpu_time << " ms" << std::endl;

    // Peak memory usage in KB. This can add quite a bit of overhead, so comment out this block if you want to measure times.
    rusage usage;
    getrusage(RUSAGE_SELF, &usage);
    std::cout << "Peak Memory Usage: " << usage.ru_maxrss << " KB" << std::endl;

    return 0;
}